hack/transform-al2-to-al2023.sh

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+
+set -o pipefail
+set -o nounset
+set -o errexit
+
+if [[ -z "${PACKER_TEMPLATE_FILE:-}" ]]; then
+  echo "PACKER_TEMPLATE_FILE must be set." >&2
+  exit 1
+fi
+if [[ -z "${PACKER_DEFAULT_VARIABLE_FILE:-}" ]]; then
+  echo "PACKER_DEFAULT_VARIABLE_FILE must be set." >&2
+  exit 1
+fi
+
+# rsa keys are not supported in al2023, switch to ed25519
+# delete the upgrade kernel provisioner as we don't need it for al2023
+cat "${PACKER_TEMPLATE_FILE}" \
+  | jq '._comment = "All template variables are enumerated here; and most variables have a default value defined in eks-worker-al2023-variables.json"' \
+  | jq '.variables.temporary_key_pair_type = "ed25519"' \
+  | jq 'del(.provisioners[5])' \
```

Comment:
We change the provisioners array often enough to not hardcode this index:
```suggestion
  | jq '.provisioners |= map(select(.script//empty|endswith("upgrade_kernel.sh")|not))' \
```